### PR TITLE
Feature/transactions

### PR DESCRIPTION
**Pull request by bashleigh**
Feel like I've introduced more issues than completed here. The goal of this PR was to implement the basic required sale, refund and find methods with types. 

### Newly created, related issues
 - Feel like the types I've added should be moved to definitelyTyped #19 
 - There should be some in depth documentation on how to use the transactions and subscription methods from start to finish including token/nonce creation #18
 - Subscription methods should be added, tested etc in a separate PR as this one is a little large #20 

### Relating issues 
 - Add transaction methods #4 
 - Add transaction types #9 
 - Add tests for transactions #6 
 - Add refund method #13 #7 (7 is duplicate) 
 
I also added my Braintree credentials to travis. Feel like maybe I should mock the responses instead of making a call the braintree. It's likely that PR's from outside of this repo will fail because travis is making the call the braintree without the creds on the contributor's repo and not this one.

### CODE REVIEW

File: src/braintree.webhook.controller.ts [hunk 1]
```diff
@@ -29,7 +29,6 @@ export default class BraintreeWebhookController {
 
   async handle(@Req() request) {
```
Comment (shekohex) [inline src/braintree.webhook.controller.ts]: this needs to be formatted, i think !

Comment (bashleigh) [reply src/braintree.webhook.controller.ts]: no I've ran format on it. What's the problem? The decorator?

Comment (bashleigh) [reply src/braintree.webhook.controller.ts]: ah think I see now!

Comment (bashleigh) [reply src/braintree.webhook.controller.ts]: ah I think see now!

**Pull request merged**